### PR TITLE
SDP: implements sticky buffers

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are checked accordingly before submitting your Pull Request -- thank you.

## Contribution style:
- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html

## Our Contribution agreements:
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)

## Changes (if applicable):
- [x] I have updated the User Guide (in [doc/userguide/](https://github.com/OISF/suricata/tree/304271e63a9e388412f25f0f94a1a0da4bf619d9/doc/userguide)) to reflect the changes made
- [x] I have updated the JSON schema (in [etc/schema.json](https://github.com/OISF/suricata/blob/304271e63a9e388412f25f0f94a1a0da4bf619d9/etc/schema.json)) to reflect all logging changes
      (including schema descriptions)
- [x] I have created a ticket at
      https://redmine.openinfosecfoundation.org/projects/suricata/issues

Link to ticket: https://redmine.openinfosecfoundation.org/issues/
- https://redmine.openinfosecfoundation.org/issues/7291
- https://redmine.openinfosecfoundation.org/issues/7305

Describe changes:
- Sticky buffers for SDP have been implemented.
- `sdp.media.bandwidth` and `sdp.media.attribute` sticky buffers have not been implemented yet because the media field is a vector of `MediaDescription`, and both the `bandwidth` and `attribute` fields are vectors as well.
I believe the multi-buffer API cannot handle such a situation.
If I'm not mistaken, a simple solution could be to "stringify" these fields like this:
```
"media_description": {
    "bandwidth": "AS:64,AS:65,AS:66",
    "attribute": "sendonly,recvonly"
}
```
- Media's encryption key is logged now.

Personal considerations:
- The `sdp.media.*` sticky buffers were initially meant to be named `sdp.media_descriptions.*`, but I realized the names were too long. I wonder if I should rename the `media_description` field to `media` for consistency.
- I don't like the name `sdp.media.media`; I would prefer to rename it to `sdp.media.name`, or just `sdp.media`, and adjust the logged field name if necessary.
 
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2076

### CODE REVIEW

File: rust/src/sdp/detect.rs [hunk 1]
```diff
@@ -0,0 +1,1219 @@
...
+    tx: *const c_void, direction: u8, buffer: *mut *const u8, buffer_len: *mut u32,
+) -> bool {
+    let tx = cast_pointer!(tx, SIPTransaction);
+    let sdp_message = match direction.into() {
+        Direction::ToServer => tx.request.as_ref().and_then(|req| req.body.as_ref()),
+        Direction::ToClient => tx.response.as_ref().and_then(|resp| resp.body.as_ref()),
+    };
+    if let Some(sdp) = sdp_message {
+        let session_name = &sdp.session_name;
+        if !session_name.is_empty() {
```
Comment (victorjulien) [inline rust/src/sdp/detect.rs:85]: should we return an empty buffer so we can match on `sdp.session_name; bsize:0;`?

Comment (glongo) [reply rust/src/sdp/detect.rs:85]: This field is mandatory, the parser will fail if it's not found.

Comment (victorjulien) [reply rust/src/sdp/detect.rs:85]: I think this code should suggest it can be empty then.

Comment (glongo) [reply rust/src/sdp/detect.rs:85]: Should a rule like `sdp.session_info; bsize:0` trigger an alert if the `session_info` field is not present at all?

Comment (catenacyber) [reply rust/src/sdp/detect.rs:85]: No, cf https://github.com/OISF/suricata/pull/11906 which adds `absent` keyword to match when the field is not present

**Comment by glongo** [comment]
Replaced with https://github.com/OISF/suricata/pull/11961

**Pull request closed without merging**